### ISSUE TITLE
DeepLinker sets Location Path after ionViewDidEnter

### ISSUE THREAD

**Issue by DominikTrenz** [driftyco/ionic#10032]
**Ionic version:**  (check one with "x")
[ ] **1.x**
[x] **2.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request

**Current behavior:**
I can't get the current path in ionViewDidEnter. When i try for example location.pathname the path of the last view is returned. When location.pathname is requested half a second later it is correct.

**Expected behavior:**
It should set the path before ionViewDidEnter is triggered. In my use case i need the current url for analytics logging

**Related code:**

```
ionViewDidEnter() {
    console.log(location.pathname);
  }
```

**Comment by jackfengji** [comment]
From source code, deeplinker's navChange is called after viewDidEnter now. So you need to get the correct pathname in the next tick.
setTimeout(() => {
  console.log(location.pathname); // this should give you correct pathname if you are not using tabs
}, 0);

However, I found that url is not correct after selecting the tabs. I have fixed it in #10067

**Comment by manucorporat** [comment]
Merged and fixed!
